<!--
screen_calibration.launch: ROS Launch file to track AR marker for robust kinect and baxter calibration.
Requirements: ar_track_alvar installed and kinect2_bridge up and running
Author: Nishanth Koganti
Date: 2016/6/15
-->

<launch>
    <!-- define args -->
    <arg name="marker_size" default="5.0" />
  	<arg name="max_track_error" default="0.2" />
  	<arg name="max_new_marker_error" default="0.08" />
  	<arg name="output_frame" default="/kinect2_link" />
  	<arg name="cam_image_topic" default="/kinect2/qhd/image_color" />
  	<arg name="cam_info_topic" default="/kinect2/qhd/camera_info" />
    <arg name="save_file"  default="$(find kinect_baxter_calibration)/files/kinect_calibration.yaml" />
    <arg name="parameters_file"  default="$(find kinect_baxter_calibration)/files/screen_calibration.yaml" />

    <!-- image to screen publisher -->
    <node pkg="baxter_examples" type="xdisplay_image.py" name="xdisplay_image" args="-f $(find kinect_baxter_calibration)/files/screen_marker.png"/>

    <!-- marker tracking node -->
    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen" args="$(arg marker_size) $(arg max_new_marker_error) $(arg max_track_error) $(arg cam_image_topic) $(arg cam_info_topic) $(arg output_frame)" />

    <!-- calibration using marker_calibration.py-->
    <node pkg="kinect_baxter_calibration" type="marker_calibration.py" name="marker_calibration" output="screen">
        <param name="save_file" value="$(arg save_file)" />
        <param name="parameters_file" value="$(arg parameters_file)" />
    </node>
</launch>
